<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Basic Genetics</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Basic Genetics</h1>



<p>This vignette will introduce you to how the basic genetic parameters like the allele frequency, the genotype frequency and Hardy-Weinberg Equilibrium results are calculated with <code>mixIndependR</code>.</p>
<div id="data-import" class="section level2">
<h2>Data Import</h2>
<p>The dataset imported should be in a format of the bi-allele data, transformed from vcf file format, Gen or other genotype data files.</p>
<pre><code>#&gt;    STR1 STR1_1 SNP1 SNP1_1
#&gt; 1    12     12    A      A
#&gt; 2    13     14    T      T
#&gt; 3    13     13    A      T
#&gt; 4    14     15    A      T
#&gt; 5    15     13    T      A
#&gt; 6    13     14    A      T
#&gt; 7    14     13    A      A
#&gt; 8    12     12    T      A
#&gt; 9    14     14    T      T
#&gt; 10   15     15    A      T</code></pre>
</div>
<div id="basic-genetic-parameters" class="section level2">
<h2>Basic Genetic Parameters</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(mixIndependR)</span></code></pre></div>
<p><code>AlleleFreq</code> calculates the allele frequencies for one dataset.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">AlleleFreq</span>(x)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt;    STR1 SNP1</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; 12  0.2  0.0</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt; 13  0.3  0.0</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt; 14  0.3  0.0</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt; 15  0.2  0.0</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt; A   0.0  0.5</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt; T   0.0  0.5</span></span></code></pre></div>
<p><code>GenotypeFreq</code> calculates the observed or expected genotype frequency. If <code>expect=FALSE</code>, the observed genotype frequencies from the original dataset will be calculated. If <code>expected=TRUE</code>, the expected genotype probabilities from allele frequency table under Hardy-Weinberg Equilibrium will be exported.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>p &lt;-<span class="st"> </span><span class="kw">AlleleFreq</span>(x)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="kw">GenotypeFreq</span>(x,p,<span class="dt">expect =</span> <span class="ot">FALSE</span>)</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt;       STR1 SNP1</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt; 12 12    2    0</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt; 13 13    1    0</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt; 14 14    1    0</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt; 15 15    1    0</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt; A A      0    2</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">#&gt; T T      0    2</span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">#&gt; 12 13    0    0</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt; 12 14    0    0</span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co">#&gt; 12 15    0    0</span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="co">#&gt; 12 A     0    0</span></span>
<span id="cb4-14"><a href="#cb4-14"></a><span class="co">#&gt; 12 T     0    0</span></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="co">#&gt; 13 14    3    0</span></span>
<span id="cb4-16"><a href="#cb4-16"></a><span class="co">#&gt; 13 15    1    0</span></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="co">#&gt; 13 A     0    0</span></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co">#&gt; 13 T     0    0</span></span>
<span id="cb4-19"><a href="#cb4-19"></a><span class="co">#&gt; 14 15    1    0</span></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="co">#&gt; 14 A     0    0</span></span>
<span id="cb4-21"><a href="#cb4-21"></a><span class="co">#&gt; 14 T     0    0</span></span>
<span id="cb4-22"><a href="#cb4-22"></a><span class="co">#&gt; 15 A     0    0</span></span>
<span id="cb4-23"><a href="#cb4-23"></a><span class="co">#&gt; 15 T     0    0</span></span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="co">#&gt; A T      0    6</span></span>
<span id="cb4-25"><a href="#cb4-25"></a><span class="kw">GenotypeFreq</span>(x,p,<span class="dt">expect =</span> <span class="ot">TRUE</span>)</span>
<span id="cb4-26"><a href="#cb4-26"></a><span class="co">#&gt;       STR1 SNP1</span></span>
<span id="cb4-27"><a href="#cb4-27"></a><span class="co">#&gt; 12 12 0.04 0.00</span></span>
<span id="cb4-28"><a href="#cb4-28"></a><span class="co">#&gt; 13 13 0.09 0.00</span></span>
<span id="cb4-29"><a href="#cb4-29"></a><span class="co">#&gt; 14 14 0.09 0.00</span></span>
<span id="cb4-30"><a href="#cb4-30"></a><span class="co">#&gt; 15 15 0.04 0.00</span></span>
<span id="cb4-31"><a href="#cb4-31"></a><span class="co">#&gt; A A   0.00 0.25</span></span>
<span id="cb4-32"><a href="#cb4-32"></a><span class="co">#&gt; T T   0.00 0.25</span></span>
<span id="cb4-33"><a href="#cb4-33"></a><span class="co">#&gt; 12 13 0.12 0.00</span></span>
<span id="cb4-34"><a href="#cb4-34"></a><span class="co">#&gt; 12 14 0.12 0.00</span></span>
<span id="cb4-35"><a href="#cb4-35"></a><span class="co">#&gt; 12 15 0.08 0.00</span></span>
<span id="cb4-36"><a href="#cb4-36"></a><span class="co">#&gt; 12 A  0.00 0.00</span></span>
<span id="cb4-37"><a href="#cb4-37"></a><span class="co">#&gt; 12 T  0.00 0.00</span></span>
<span id="cb4-38"><a href="#cb4-38"></a><span class="co">#&gt; 13 14 0.18 0.00</span></span>
<span id="cb4-39"><a href="#cb4-39"></a><span class="co">#&gt; 13 15 0.12 0.00</span></span>
<span id="cb4-40"><a href="#cb4-40"></a><span class="co">#&gt; 13 A  0.00 0.00</span></span>
<span id="cb4-41"><a href="#cb4-41"></a><span class="co">#&gt; 13 T  0.00 0.00</span></span>
<span id="cb4-42"><a href="#cb4-42"></a><span class="co">#&gt; 14 15 0.12 0.00</span></span>
<span id="cb4-43"><a href="#cb4-43"></a><span class="co">#&gt; 14 A  0.00 0.00</span></span>
<span id="cb4-44"><a href="#cb4-44"></a><span class="co">#&gt; 14 T  0.00 0.00</span></span>
<span id="cb4-45"><a href="#cb4-45"></a><span class="co">#&gt; 15 A  0.00 0.00</span></span>
<span id="cb4-46"><a href="#cb4-46"></a><span class="co">#&gt; 15 T  0.00 0.00</span></span>
<span id="cb4-47"><a href="#cb4-47"></a><span class="co">#&gt; A T   0.00 0.50</span></span></code></pre></div>
<p><code>Heterozygous</code> test the heterozygosity of each individuals at each locus and output a table with <code>0</code> denoting homozygous and <code>1</code> heterozygous.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>h &lt;-<span class="kw">Heterozygous</span>(x)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">print</span>(h)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt;       STR1 SNP1</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt;  [1,]    0    0</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt;  [2,]    1    0</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt;  [3,]    0    1</span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt;  [4,]    1    1</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt;  [5,]    1    1</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt;  [6,]    1    1</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt;  [7,]    1    0</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt;  [8,]    0    1</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt;  [9,]    0    0</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; [10,]    0    1</span></span></code></pre></div>
<p><code>RxpHetero</code> calculate Real or Expected Average Heterozygosity at each locus. If <code>HWE=TRUE</code>, this function will calculate the expected heterozygosities under Hardy-Weinberg Equilibrium; If <code>HWE=FALSE</code>, this function will calculate the real average heterozygosities.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>H &lt;-<span class="st"> </span><span class="kw">RxpHetero</span>(h,p,<span class="dt">HWE=</span><span class="ot">TRUE</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">head</span>(H)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt; STR1 SNP1 </span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#&gt; 0.74 0.50</span></span></code></pre></div>
<p><code>AlleleShare_Table</code> calculates the table of number of shared alleles for each pair of individuals at each locus.If <code>replicate=TRUE</code>, the pairs are formed with replicates; if <code>replicate=FALSE</code>, the pairs are formed without replicate.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>AS&lt;-<span class="kw">AlleleShare_Table</span>(x,<span class="dt">replicate=</span><span class="ot">TRUE</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">head</span>(AS)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt;      STR1 SNP1</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; 1vs2    0    0</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt; 1vs3    0    1</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; 1vs4    0    1</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; 1vs5    0    1</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; 1vs6    0    1</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; 1vs7    0    2</span></span></code></pre></div>
<p><code>RealProAlleleShare</code> and <code>ExpProAllelShare</code> calculate the average proportions and the expected probabilities of sharing 0,1 and 2 alleles at each locus.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>e &lt;-<span class="kw">RealProAlleleShare</span>(AS)</span>
<span id="cb8-2"><a href="#cb8-2"></a>e0&lt;-<span class="kw">ExpProAlleleShare</span>(p)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw">head</span>(e)</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#&gt;              P0        P1         P2</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt; STR1 0.53333333 0.3777778 0.08888889</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt; SNP1 0.08888889 0.5333333 0.37777778</span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="kw">head</span>(e0)</span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt;         P0     P1     P2</span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt; STR1 0.317 0.5672 0.1158</span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; SNP1 0.125 0.5000 0.3750</span></span></code></pre></div>
<p><code>HWE_Chisq</code> and <code>HWE_Fisehr</code> test the Hardy-Weinberg Equilibrium with Pearson’s Chi-square test or Fisher’s exact test. <code>B</code> is an integer specifying the number of replicates used in the Monte Carlo test.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>g &lt;-<span class="st"> </span><span class="kw">GenotypeFreq</span>(x,p,<span class="dt">expect=</span><span class="ot">FALSE</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a>g0 &lt;-<span class="st"> </span><span class="kw">GenotypeFreq</span>(x,p,<span class="dt">expect=</span><span class="ot">TRUE</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="kw">HWE.Chisq</span>(g,g0,<span class="dt">rescale.p=</span><span class="ot">FALSE</span>,<span class="dt">simulate.p.value=</span><span class="ot">TRUE</span>,<span class="dv">2000</span>)</span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co">#&gt; $chi</span></span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="co">#&gt;     STR1     SNP1 </span></span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co">#&gt; 4.544444 0.400000 </span></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="co">#&gt; </span></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#&gt; $pvalue</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#&gt;      STR1      SNP1 </span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#&gt; 0.6266867 0.8515742</span></span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="kw">HWE.Fisher</span>(p,H,g<span class="op">/</span><span class="kw">colSums</span>(g))</span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co">#&gt;      STR1      SNP1 </span></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="co">#&gt; 0.8173266 0.9387241</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
